index.bs

```diff
@@ -1004,6 +1005,26 @@ spec: webidl
       asynchronous methods in the [[clipboard-apis]].
     </p>
   </section>
+  <section>
+    <h3 id="screen-capture">
+      Screen Capture
+    </h3>
+    <p>
+      The <dfn for="PermissionName" enum-value>"display"</dfn>
+      permission is the permission associated with the usage of
+      [[screen-capture]].
+    </p>
+    <dl>
+      <dt>
+        <a>permission state constraints</a>
+      </dt>
+      <dd>
+        Valid values for this descriptor's <a>permission state</a> are
+        {{"prompt"}} and {{"denied"}}. The user agent MUST NOT ever set this
```

Comment:
Why should the spec *require* the permission to never be set as granted? Is it a design requirement?

Reply:
Agree. This might be overstepping.

Reply:
Yes, it's a design requirement, due to [security risks](https://tools.ietf.org/html/draft-ietf-rtcweb-security-10#section-4.1.1) inherent with screen-capture.

[RTCWEB-SECURITY-ARCH](https://tools.ietf.org/html/draft-ietf-rtcweb-security-arch-15#section-5.2) says: *"Browsers MUST NOT permit permanent screen or application sharing permissions to be installed as a response to a JS request for permissions."*.

Reply:
A [blog explainer] (https://blog.mozilla.org/webrtc/share-browser-windows-entire-screen-sites-trust/).

Reply:
The [spec prose](https://w3c.github.io/mediacapture-screen-share/#dom-navigator-getdisplaymedia) is already pretty clear, saying: *"The user agent MUST let the end-user choose which display surface to share out of all available choices every time"*.

Adding this language here is necessary to for that spec to call this spec's [prompt the user to choose ](https://w3c.github.io/permissions/#prompt-the-user-to-choose) algorithm, since that algorithm is not guaranteed to prompt the user otherwise.

Reply:
Ok, but this reads like this can never become "granted". That's not the case tho. Maybe this this to be phrased that it can only be "granted" by explicit user consent.

Reply:
I think we're fine. I think you're confusing algorithms that *return* `"granted"`, which is fine, with setting the *state*, which is not.

Reply:
Could be. Admittedly, not looked at this in a while.

Reply:
The [prompt the user to choose ](https://w3c.github.io/permissions/#prompt-the-user-to-choose) algorithm *"returns either `"denied"` or one of the options."*